Makefile

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 The kube-storage Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SHELL:=/bin/bash
+
+PROTOC_REQ := "libprotoc 3.14.0"
+PROTOC_GEN_GO_REQ := "protoc-gen-go v1.25.0"
+PROTOC_GEN_GO_GRPC_REQ := "protoc-gen-go-grpc 1.1.0"
+
+PROTOC := $(shell ([ "`$(PWD)/protoc --version 2> /dev/null`" == $(PROTOC_REQ) ] && echo true))
+PROTOC_GEN_GO := $(shell ([ "`$(PWD)/protoc-gen-go --version 2> /dev/null`" == $(PROTOC_GEN_GO_REQ) ] && echo true))
+PROTOC_GEN_GO_GRPC := $(shell ([ "`$(PWD)/protoc-gen-go-grpc --version 2> /dev/null`" == $(PROTOC_GEN_GO_GRPC_REQ) ] && echo true))
+
+all: install-deps build
+
+build: install-deps
+	# generate libs
+	./protoc --go_out=lib/go/replication --go_opt=paths=source_relative --plugin=./protoc-gen-go *.proto
+	./protoc --go-grpc_out=lib/go/replication --go-grpc_opt=paths=source_relative --plugin=./protoc-gen-go-grpc *.proto
+
+install-deps:
+ifndef PROTOC
+	# download protoc
+	curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.14.0/protoc-3.14.0-linux-x86_64.zip
```

Comment:
```suggestion
curl -OL https://github.com/protocolbuffers/protobuf/releases/download/${PROTOC_VERSION}/protoc-3.14.0-linux-x86_64.zip
```

Reply:
for all the tools we installed the version should be defined as a variable above and we should use the same here.